flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/DbSupport.java

```diff
@@ -144,6 +143,30 @@ public void restoreCurrentSchema() {
     protected abstract void doChangeCurrentSchemaTo(String schema) throws SQLException;
 
     /**
+     * Retrieves the current user.
+     *
+     * @return The current user for this connection.
+     */
+    public String getCurrentUserName() {
```

Comment:
If you meant the public method `getCurrentUserName`, not its protected implementation method `doGetCurrentUserName`, then the answer to you question is - this public method should be a part of Flyway API, just as all other public methods of this class. There may be different use cases for it.